### PR TITLE
Implement docker module

### PR DESCRIPTION
**Pull request by pawloch00**
## Fixes / Features
Implement module to interact with cluster toolkit via docker. It should allow to copy deployment blueprint into deployment directory mounted to container in which gcluster command is run.

## Testing / Documentation
Testing details.

- [ y/n ] Tests pass
- [ y/n ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: src/xpk/core/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""
+Copyright 2024 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     https://www.apache.org/licenses/LICENSE-2.0
```
Comment (mbobrovskyi) [inline src/xpk/core/tests/__init__.py:6]:
```suggestion
You may obtain a copy of the License at
     
     https://www.apache.org/licenses/LICENSE-2.0
     
```

File: src/xpk/core/tests/test_docker_manager.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+"""
+Copyright 2024 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     https://www.apache.org/licenses/LICENSE-2.0
```
Comment (mbobrovskyi) [inline src/xpk/core/tests/test_docker_manager.py:6]:
```suggestion
     
     https://www.apache.org/licenses/LICENSE-2.0
     
```

File: src/xpk/core/docker_manager.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    return True
+
+  def build_image(
+      self, img_name: str = xpk_ctk_img_name, nocache: bool = False
+  ):
+    """Build image from dockerfile pointed by _img_name. This method
+    uses python docker client to build cloud toolkit execution image.
+    Arguments:
+    Returns:
+      - None
```
[the remark this answers is not in the document]
Comment (pawloch00) [reply src/xpk/core/docker_manager.py:65]: I don't see why

[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/core/docker_manager.py:65]: Sorry. Here not need.

**Review by mbobrovskyi — APPROVED** [review]

**Pull request merged**